lib/mongo/auth/stringprep.rb

```diff
@@ -2,9 +2,29 @@
 
 module Mongo
   module Auth
+    # This namespace contains all behavior related to string preparation (RFC 3454).
+    #
+    # @since 2.6.0
     module StringPrep
       extend self
 
+      # Prepare a string given a set of mappings and prohibited character tables.
+      #
+      # @example Prepare a string.
+      #   StringPrep.prepare("some string",
+      #                      StringPrep::SASL::Mappings,
+      #                      StringPrep::SASL::Prohibited,
+      #                      normalize: true, bidi: true)
+      #
+      # @param [ String ] data The string to prepare.
+      # @param [ Array ] mappings A list of mappings to apply to the data.
+      # @param [ Array ] prohibited A list of prohibited character lists to ensure the data doesn't
+      #   contain after mapping and normalizing the data. If the mapped and normalized data contains
+      #   a character in one of the lists, prepare will raise an error.
+      # @param [ Hash ] options Optional operations to perform during string preparation. Valid keys
+      #   are :normalize and :bidi.
+      #
```

Comment:
Usually we document the specific options that are available as well and don't write what they are in-line. Check out another file in the driver for the syntax.